exonum-java-binding-core/rust/src/utils/time_service.rs

```diff
@@ -12,123 +12,16 @@
 
 #[no_mangle]
 pub extern "system" fn Java_com_exonum_binding_time_TimeSchemaProxy_isTimeServiceEnabled(
     _: JNIEnv,
     _: JClass,
 ) -> jboolean {
-    is_time_service_enabled().into()
-}
-
-/// Initializes static context with reference to blockchain
-pub fn init_with_blockchain(blockchain: Blockchain) {
-    unsafe {
-        INIT.call_once(|| BLOCKCHAIN = Some(blockchain));
-    }
-}
-
-fn is_time_service_enabled() -> bool {
-    INIT.state().done()
-        && unsafe { BLOCKCHAIN.clone() }.map_or(false, |blockchain| {
-            blockchain.service_map().get(&TIME_SERVICE_ID).is_some()
-        })
-}
-
-#[cfg(test)]
-mod tests {
-    extern crate exonum_time;
-    extern crate futures;
-
-    use self::futures::sync::mpsc;
-    use super::*;
-    use exonum::{
-        blockchain::{Service, Transaction},
-        crypto::{gen_keypair, Hash},
-        messages::RawTransaction,
-        node::ApiSender,
-        storage::{MemoryDB, Snapshot},
-    };
-
-    struct EmptyService;
-
-    impl Service for EmptyService {
-        fn service_id(&self) -> u16 {
-            0
-        }
-
-        fn service_name(&self) -> &str {
-            "empty_service"
-        }
-
-        fn state_hash(&self, _: &Snapshot) -> Vec<Hash> {
-            vec![]
-        }
-
-        fn tx_from_raw(&self, _: RawTransaction) -> Result<Box<dyn Transaction>, failure::Error> {
-            unimplemented!()
-        }
-    }
-
-    struct StubTimeService;
-
-    impl Service for StubTimeService {
-        fn service_id(&self) -> u16 {
-            TIME_SERVICE_ID
-        }
-
-        fn service_name(&self) -> &str {
-            "time_service"
-        }
-
-        fn state_hash(&self, _: &Snapshot) -> Vec<Hash> {
-            vec![]
-        }
-
-        fn tx_from_raw(&self, _: RawTransaction) -> Result<Box<dyn Transaction>, failure::Error> {
-            unimplemented!()
-        }
-    }
-
-    fn create_blockchain(services: Vec<Box<dyn Service>>) -> Blockchain {
-        let (public_key, secret_key) = gen_keypair();
-        let api_channel = mpsc::channel(128);
-        let app_tx = ApiSender::new(api_channel.0);
-        let storage = MemoryDB::new();
-        let blockchain = Blockchain::new(storage, services, public_key, secret_key, app_tx.clone());
-        blockchain
-    }
-
-    #[test]
-    #[ignore]
-    fn is_not_initialised_with_blockchain() {
-        assert!(!is_time_service_enabled());
-    }
-
-    #[test]
-    #[ignore]
-    fn initialized_no_time_service() {
-        let blockchain = create_blockchain(vec![Box::new(EmptyService)]);
-        init_with_blockchain(blockchain);
-        assert!(!is_time_service_enabled());
-    }
-
-    #[test]
-    fn initialized_time_service_presented() {
-        let blockchain = create_blockchain(vec![Box::new(EmptyService), Box::new(StubTimeService)]);
-        init_with_blockchain(blockchain);
-        assert!(is_time_service_enabled());
-    }
+    is_service_enabled(TIME_SERVICE).into()
```

Comment:
Is this code seriously loads a file on each invocation (each time a TimeSchema is accessed)?

Reply:
Yes, this is a heritage of previous checking from runtime )). Thanks for pointing this out - corrected.